src/isomorphic/classic/element/ReactElement.js

```diff
@@ -194,12 +186,28 @@ ReactElement.createElement = function(type, config, children) {
   // the newly allocated props object.
   var childrenLength = arguments.length - 2;
   if (childrenLength === 1) {
-    props.children = children;
+    if (Array.isArray(children)) {
+      let childArray = children.slice(0);
```

Comment:
I think it's important that the behavior stays the same across different envs so I call `slice` here regardless of the environment (prod vs. dev). Without this we won't be able to catch cases like `var children = [...]; <Comp children={children}>` since we don't want to freeze the original array reference in userland.